gateway/src/apicast/policy/token_introspection/token_introspection.lua

```diff
@@ -26,12 +28,28 @@ function _M.new(config)
       ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') }
     }
   }
+
+  -- If there's no 'max_ttl_tokens', the ttl will be the one returned by the
+  -- token introspection endpoint. To disable caching, max_ttl_tokens needs to
+  -- be set to 0.
+  self.caching_enabled = not config.max_ttl_tokens or config.max_ttl_tokens > 0
```

Comment:
What about:

```lua
local max_cached_tokens = tonumber(config.max_cached_tokens) or 0

self.caching_enabled = max_cached_tokens > 0
```

Reply:
The advantage of your solution is that when none of the 2 new config params are specified, the policy behaves as it does now, that is, without caching anything.

Reply:
Changed :+1: